tests/queue_timeout.cpp

```diff
@@ -115,6 +115,50 @@ static void test_queue_timeout(session &s, const nodes_data *setup) {
 	}
 }
 
+/* The test checks for error reply on dropped request, so client could be informed on timeout earlier,
+ * not on session timeout.
+ */
+static void test_queue_ack_timeout(session &s, const nodes_data *setup) {
+	// check that test have only one node
+	BOOST_REQUIRE(setup->nodes.size() == 1);
+
+	// test key and data
+	std::string key = "queue timeout reply with timout test key";
+	std::string data = "queue timeout reply with timout test key";
+
+	// write test key/data
+	ELLIPTICS_REQUIRE(async_write, s.write_data(key, data, 0));
+
+	const auto &node = setup->nodes.front();
+	// sets 2 seconds delay to the only backend on the only node
+	constexpr uint64_t delay_ms = 2000;
+	s.set_delay(node.remote(), backend_id, delay_ms).get();
+
+	// sets 10 seconds session timeout - it should fit at least 2 x backend delay because
+	// if the second command won't be dropped due to queue timeout its handling time
+	// should be around 4 seconds (2 x backend delay).
+	constexpr auto session_timeout = 10;
+	s.set_timeout(session_timeout);
+	dnet_time start; dnet_current_time(&start);
+
+	// first lookup will delay after decoupling from queue for `delay_ms`.
+	auto async = s.lookup(key);
+	// second lookup command. It will be in io queue while the only io thread will sleep on backend delay.
+	auto async_timeouted = s.lookup(key);
+	{
+		ELLIPTICS_CHECK(res, std::move(async));
+	} {
+		// second lookup command should faile with timeout error due to message expiration.
+		ELLIPTICS_REQUIRE_ERROR(res, std::move(async_timeouted), -ETIMEDOUT);
```

Comment:
Could we also expect and verify that result also has `DNET_FLAGS_REPLY` in its flags?

Reply:
It seems not possible, `callback_result_entry::is_valid()` is false on response error and it is exception thrown on `callback_result_entry::command()` access attempt. 

I've tried with different exception policies in session instance.